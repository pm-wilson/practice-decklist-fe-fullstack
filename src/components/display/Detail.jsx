import React from 'react';
import PropTypes from 'prop-types';

const Detail = ({ deck, colors, format, handleChange, handleSubmit }) => {
  return (
    <div>
      <form onSubmit={handleSubmit}>
        <h2>Deck Details</h2> 
        <label>Deck:
          <input name="deck" onChange={handleChange} value={deck}></input>
        </label>
        <label>Colors:
          <input name="colors" onChange={handleChange} value={colors}></input>
        </label>
        <label>Format:
          <input name="format" onChange={handleChange} value={format}></input>
        </label>
        <button>Update</button>
      </form>
      <button>Delete</button>
    </div>
  );
};

Detail.propTypes = {
  deck: PropTypes.string,
  colors: PropTypes.string,
  format: PropTypes.string,
  handleChange: PropTypes.func,
  handleSubmit: PropTypes.func,
};

export default Detail;
