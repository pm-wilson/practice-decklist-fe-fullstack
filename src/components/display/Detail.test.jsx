import React from 'react';
import { render } from '@testing-library/react';
import Detail from './Detail';

describe('Detail component', () => {
  const details = { name: 'tim', photoUrl: 'www.google.com', eye: 'brown', hair: 'black', weapon: 'wind' };

  it('renders Detail', () => {
    const { asFragment } = render(<Detail details={details} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
