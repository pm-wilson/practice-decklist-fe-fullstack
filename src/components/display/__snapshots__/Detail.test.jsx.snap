// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail component renders Detail 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h2>
        Deck Details
      </h2>
      <label>
        Deck:
        <input
          name="deck"
          value=""
        />
      </label>
      <label>
        Colors:
        <input
          name="colors"
          value=""
        />
      </label>
      <label>
        Format:
        <input
          name="format"
          value=""
        />
      </label>
      <button>
        Update
      </button>
    </form>
    <button>
      Delete
    </button>
  </div>
</DocumentFragment>
`;
